src/Sarif/Readers/RuleDictionaryConverter.cs

```diff
@@ -6,9 +6,13 @@
 
 namespace Microsoft.CodeAnalysis.Sarif.Readers
 {
-    public class IRuleDictionaryConverter : JsonConverter
+    /// <summary>
+    /// This converter only exists because the version of Json.NET we're using (6.0.8)
+    /// can't write out a property whose static type is an interface.
+    /// </summary>
+    public class RuleDictionaryConverter : JsonConverter
     {
-        public static readonly IRuleDictionaryConverter Instance = new IRuleDictionaryConverter();
+        public static readonly RuleDictionaryConverter Instance = new RuleDictionaryConverter();
```

Comment:
I think the real (verbose) convention is to specify both the key and value, so StringIRuleDictionaryConverter would be the name. 'RuleDictionaryConverter' is fine with me, though (and preferable to previous.

---
In reply to: [63249962](https://github.com/Microsoft/sarif-sdk/pull/270#discussion_r63249962) [](ancestors = 63249962)